bootstrap/container/dev_or_remote_mode.go

```diff
@@ -0,0 +1,36 @@
+/*******************************************************************************
+ * Copyright 2023 Intel Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+package container
+
+import (
+	"github.com/edgexfoundry/go-mod-bootstrap/v3/di"
+)
+
+type InDevOrRemoteMode struct {
+	Value bool
+}
+
+// DevOrRemoteModeName contains the name of the Dev or Remotes mode flag in the DIC.
+var DevOrRemoteModeName = di.TypeInstanceToName((*InDevOrRemoteMode)(nil))
+
+// DevOrRemoteModeFrom helper function queries the DIC and returns the Dev or Remotes mode flag.
+func DevOrRemoteModeFrom(get di.Get) bool {
```

Comment:
Since it's the public di package, any future change may become a breaking change.
Do we always bundle Dev and Remote mode?  For the other service implementation, they might just need to confirm it's Dev mode or Remote mode.  I suggest to return a simple Service Mode struct contains `IsDev` and `IsRemote`, or implement two separated di container for `DevModeFrom` and `RemoteModeFrom`/`RemoteHostsFrom`.

Reply:
Good point. I like the structure approach. Will make that change.